resources/type-system/flow-analysis.md

```diff
@@ -15,7 +15,46 @@ reachability analysis.
 
 ## Motivation
 
-TODO
+The Dart language spec requires type promotion analysis to determine when
+a local variable is known to have a more specific type than its declared type.
+This is currently performed by the front end and the analyzer which perform additional
+"type propagation" as outlined in the [Enhanced Type Promotion](
+https://github.com/dart-lang/language/blob/master/working/enhanced-type-promotion/feature-specification.md)
+proposal (see [tracking issue](https://github.com/dart-lang/language/issues/81)). We believe
+additional enhancement is necessary beyond this proposal to support NNBD (“non-nullability by default”).
+
+The language spec also requires a small amount of reachability analysis,
+to ensure control flow does not reach the end of a switch case.
+To support NNBD a more complete form of reachability analysis is necessary to detect
+when control flow “falls through” to the end of a function body and an implicit `return null;`,
+which would be an error in a function with a non-nullable return type.
+
+TODO: Links to proposals or issues requesting additional reachability analysis?
```

Comment:
This is the issue I'm aware of: https://github.com/dart-lang/language/issues/139

Reply:
Added.